src/entities/repos/submission_repo.py

```diff
@@ -42,29 +46,44 @@ async def get_latest_submission(session: AsyncSession, filing_id: int) -> List[S
 
 
 async def get_filing_periods(session: AsyncSession) -> List[FilingPeriodDAO]:
-    async with session.begin():
-        stmt = select(FilingPeriodDAO)
-        results = await session.scalars(stmt)
-        return results.all()
+    return await query_helper(session, FilingPeriodDAO)
 
 
 async def get_submission(session: AsyncSession, submission_id: int) -> SubmissionDAO:
-    return await query_helper(session, submission_id, SubmissionDAO)
+    result = await query_helper(session, SubmissionDAO, "id", submission_id)
+    return result[0] if result else None
 
 
 async def get_filing(session: AsyncSession, filing_id: int) -> FilingDAO:
-    return await query_helper(session, filing_id, FilingDAO)
+    result = await query_helper(session, FilingDAO, "id", filing_id)
+    result = deepcopy(result)
+    if result:
+        await populate_missing_tasks(session, result)
```

Comment:
Maybe a little pedantic, but the `deepcopy` call can probably happen within the if statement.
Another thing, if we can, we'd like to avoid side effects, this could be as simple as the `populate_missing_tasks` method returning the object. so `result = await populate_missing....()`.
Last bit, just checking the method, the `filing_tasks` retrieval can probably be cached since there shouldn't be frequent changes if any to that table, so you'd abstract out that portion of code into a separate function. I know sqlalchemy would cache the query anyway, but the async_lru would let us cache it without hitting the sqlalchemy code.

Reply:
Updated based on our conversation (moving the deepcopy and returning the new list), also pulled in async_lru.  Ran through some tests over 200 iterations and without the decorator each fetch was taking on the order of 10-15 milliseconds.  With the decorator, first fetch took that long but subsequent fetches took about 25 microseconds.  So thanks for introducing me to lru caching, was not aware of it but definitely something to keep in mind.